bin/node/cli/src/service.rs

```diff
@@ -54,6 +55,8 @@ macro_rules! new_full_start {
 		type RpcExtension = jsonrpc_core::IoHandler<sc_rpc::Metadata>;
 		let mut import_setup = None;
 		let inherent_data_providers = sp_inherents::InherentDataProviders::new();
+		// WIP: sort out construction
+		let shared_voter_state = Arc::new(RwLock::new(None));
```

Comment:
This part here, how to construct when we don't yet have `Environment` available. Any ideas on the best approach?

Reply:
I have no idea what `VoterState` is.

Reply:
tried checking out this PR, but it looks like you're using a custom finality-grandpa crate. Try using a git dependency.

Reply:
Yeah that's right, it depends on https://github.com/paritytech/finality-grandpa/pull/114

Reply:
Updated to use git dependency pointing to my fork on github

Reply:
This object will always be constructed by the Voter in finality-grandpa. You just keep it uninitialized here.